File: app/src/main/java/abzalov/ruslan/pocketdoc/BaseContract.java
package abzalov.ruslan.pocketdoc;

import android.view.Menu;

/**
 * Интерфейс, описывающий основной контракт паттерна MVP.
 */
public interface BaseContract {

    interface BaseView {

        void setOptionsMenuVisible(Menu menu, boolean isVisible);

        void showErrorDialog(Throwable throwable);

        void showProgressBar();

        void hideProgressBar();

        void hideRefreshing();
    }

    interface BasePresenter<T extends BaseView> {

        void attachView(T t);

        void detachView();
    }
}

File: app/src/main/java/abzalov/ruslan/pocketdoc/BaseContract.kt
package abzalov.ruslan.pocketdoc

import android.view.Menu

/**
 * Интерфейс, описывающий основной контракт.
 */
interface BaseContract {

    interface BaseView {

        fun setOptionsMenuVisible(menu: Menu, isVisible: Boolean)

        fun showErrorDialog(throwable: Throwable)

        fun showProgressBar()

        fun hideProgressBar()

        fun hideRefreshing()
    }

    interface BasePresenter<in T : BaseView> {

        fun attachView(t: T)

        fun detachView()
    }
}